### ISSUE TITLE
IPv4 address not displayed in GUI after boot on PPoE connection

### ISSUE THREAD

**Issue by mickeyreg** [ericpaulbishop/gargoyle#925]
Gargoyle 1.13pre9 from eko.one.pl, tested on 1043ND and on Archer C7.

I've done the following things:
- I configured the router with PPPoE to WAN
- I disconnected the router from the power supply
- I turned it on again
- I looked at /overview.sh and there was an IPv6 address in the "WAN IP Address:" field
- I logged in via SSH
- I did /etc/init.d/network restart
- I looked at the Status page again and then there was the correct IPv4 address

But: Before I've restarted the network, I've checked connection - both the router and my machine were connected to the Internet. OpenVPN clients were connected. The ifconfig command displayed a valid IPv4 address.

So my conclusion is as follows: It looks like everything was working fine, only the Gargoyle interface did not show the IPv4 address assigned via PPPoE.

**Comment by lantis1008** [comment]
Can you please reproduce the issue (get it back to where the address is missing) and provide the following output:
```
ubus call network.interface dump
gargoyle_header_footer -i
```

Thanks

**Comment by lantis1008** [comment]
Please also provide the same output after running the network restart.

**Comment by mickeyreg** [comment]
[status.zip](https://github.com/ericpaulbishop/gargoyle/files/6330952/status.zip)

When I wanted to reproduce problem I made power cycle 3 times, and only after the third attempt problem appeard again :/

In the archive:
- after_boot_1.txt: after boot, WAN IP ok
- after_boot_2.txt: after boot, WAN IPv4 not displayed
- network_restart.txt: after network restart, when IPv4 was not visible after power cycle

**Comment by lantis1008** [comment]
Thanks for that. It quite clearly shows the problem. How it has actually happened though I am not entirely sure. I'll try to look into this.

**Comment by lantis1008** [comment]
I have not been able to reproduce this issue unfortunately.
If i had to guess, it is possibly a memory re-use or pointer type error.

If anyone has the ability to provide further insight or test cases that produce the issue 100% of the time, i'd be happy to try again.

**Comment by mickeyreg** [comment]
As I wrote before, the problem is also not 100% reproducible in my case :( But when I saw, that it is only problem with data displayed I'm ignoring it ;)

I'm writing now, because last time I found probably the same problem but in a little different situation. My English is rather poor, but I'll try to describe:

I wanted to make "a duplicate" of my main router. So I made a backup and restored it on the second device. But I made it in the following steps:
- I connected the new router (with default Gargoyle configuration) between my PC and my local network, so on the LAN interface it has 192.168.1.1 and on the WAN the router received IP from my home network via DHCP (from the main router)
- next I restored configuration of my main router to the new one. and the device was rebooted

After this operation, my new router had:
- on LAN IP of the main router - 192.168.32.1 - the PC was reconfigured by DHCP
- on WAN ... nothing, because it was configured to use PPoE, but my home network does not have the server configured

After all I entered the web interface of the new router to check settings. I saw that WAN address is configured ... it was IP assigned by DHCP, from time when the router had default Gargoyle configuration. I don't know how this information could be generated: the device was rebooted, after reboot it should not contact with DHCP server, so it was from ... ??

**Comment by obsy** [comment]
@lantis1008

Next problem: see wan ip format:

![alt text](https://images92.fotosik.pl/510/4bce0322b6955e97.png)

More details: https://eko.one.pl/forum/viewtopic.php?pid=255560#p255560

**Comment by lantis1008** [comment]
I will patch something that will fix another issue (which will hide this one), but doesn't fix the real problem.
Can i get the output of `cat /proc/net/route` from the user please?

**Comment by lantis1008** [comment]
I have a feeling that https://github.com/ericpaulbishop/gargoyle/commit/faa424eabbe4ed3d854c2eb6d4c1fbd8c68dc7e5 will help this issue. Can you please check again @mickeyreg

**Comment by mickeyreg** [comment]
Check... How...

OK. I cloned gargoyle repo to my VM with Ubuntu and ran: `make ath79 FULL_BUILD=true` and then I copied `gargoyle_header_footer` file to my router. Should it be enough?

With old file I had:
![old_ghf](https://user-images.githubusercontent.com/5645076/122772322-8ce32580-d2a7-11eb-85ad-2da16024adbe.png)

And withe the new one:
![new_ghf](https://user-images.githubusercontent.com/5645076/122772367-97052400-d2a7-11eb-8b9a-630dd292cd47.png)

I'm not sure what should be in _WAN MAC Address_ filed on PPPoE connection.

**Comment by lantis1008** [comment]
No logic should have changed with the MAC address check, so it's surprising that it has disappeared.
Have you tried rebooting? 
Can you provide these again?
```
ubus call network.interface dump
gargoyle_header_footer -i
```

**Comment by mickeyreg** [comment]
```
root@gate:~# uptime
 20:03:38 up 50 days, 20:28,  load average: 0.00, 0.02, 0.00
```

There is no MAC, so maybe all zeros is ok for PPPoE?

```
root@gate:~# ifconfig
pppoe-wan Link encap:Point-to-Point Protocol
          inet addr:134.220.32.23  P-t-P:10.0.0.23  Mask:255.255.255.255
          inet6 addr: fe80::9d21:c18d:a112:ea8/10 Scope:Link
          UP POINTOPOINT RUNNING NOARP MULTICAST  MTU:1480  Metric:1
          RX packets:71834221 errors:0 dropped:0 overruns:0 frame:0
          TX packets:73260696 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:3
          RX bytes:20580204817 (19.1 GiB)  TX bytes:16816301975 (15.6 GiB)
```

The rest in attached file.
[info.txt](https://github.com/ericpaulbishop/gargoyle/files/6696479/info.txt)

**Comment by lantis1008** [comment]
I'll look into this. The mac address should be valid so something has gone awry.

**Comment by lantis1008** [comment]
This issue should be fixed.
If you can confirm and then close it that would be great thanks :)
https://github.com/ericpaulbishop/gargoyle/commit/6326b831cfe447feabec0040e7095e12bfc95197

**Comment by mickeyreg** [comment]
Works. Thanks. :)
